src/coq_rules.ml

```diff
@@ -90,11 +120,42 @@ let create_ccoq sctx ~dir =
   ; coqpp  = rr "coqpp"
   }
 
+(* compute include flags and mlpack rules *)
+let setup_ml_deps ~scope ~stdlib_dir libs =
+
+  (* coqdep expects an mlpack file next to the sources otherwise it
+   * will omit the cmxs deps *)
+  let ml_pack_files lib =
+    let plugins = Mode.Dict.get (Lib.plugins lib) Mode.Native in
+    (* Should clean up once we can depend on >= 4.04 *)
+    let to_mlpack file =
+      Path.(of_string ((fst @@ Filename.split_extension (to_string file)) ^ ".mlpack")) in
+    List.map plugins ~f:to_mlpack
+  in
+
+  (* Pair of include flags and paths to mlpack *)
+  let ml_iflags, mlpack =
+    let lib_db = Scope.libs scope in
+    let loc = Loc.none in
```

Comment:
Can use the location from the list of dependencies?

Reply:
Sure, now you reminded me that a whole pass on error handling may be merited